service/src/main/java/bio/terra/landingzone/service/landingzone/azure/LandingZoneService.java

```diff
@@ -111,6 +116,56 @@ public void deleteLandingZone(String landingZoneId) {
     throw new LandingZoneDeleteNotImplemented("Delete operation is not implemented");
   }
 
+  public LandingZoneResourcesByPurpose listResourcesWithPurposes(
+      AzureCloudContext azureCloudContext) {
+    LandingZoneManager landingZoneManager =
+        landingZoneManagerProvider.createLandingZoneManager(azureCloudContext);
+
+    return LandingZoneResourcesByPurpose.builder()
+        .generalResources(listGeneralResourcesWithPurposes(landingZoneManager))
+        .subnetResources(listVNetResourcesWithPurposes(landingZoneManager))
+        .build();
+  }
+
+  private Map<String, List<LandingZoneResource>> listGeneralResourcesWithPurposes(
```

Comment:
Since we have a `ResourcePurpose` type, should we use that for the key instead of `String`?

Reply:
Fixed